### PR TITLE
Check issues for existing sources

### LINKED ISSUE #41: Check for existing sources when requested
### Expected behavior

- User creates a new source request issue in https://github.com/tachiyomiorg/tachiyomi-extensions/, including the URL
- Action checks if new issue is a source request
- If yes, then check repo `index.min.json` for duplicate URL.
  - If duplicate found, close issue and abort action
  - If no duplicate found, continue
- If no, continue

### PR DESCRIPTION
**Pull request by alessandrojean**
- Add existing sources check by fetching the repo JSON.
- Make the duplicate comment body configurable.

Still needs some tests.

Closes #41.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -15,7 +15,7 @@
   "license": "MPL-2.0",
   "devDependencies": {
     "@types/dedent": "0.7.0",
-    "@types/node": "16.11.7",
+    "@types/node": "18.0.6",
```
Comment (arkon) [inline package.json:18]: We actually use Node 16: https://github.com/tachiyomiorg/issue-moderator-action/blob/5e482873a3ccc54aa44865816fc42a8d7c24721a/action.yml#L49

File: src/feature/dupe-check.ts [hunk 2]
```diff
@@ -1,12 +1,11 @@
...
 import * as github from '@actions/github';
 import { Issue, IssuesOpenedEvent } from '@octokit/webhooks-definitions/schema';
-import dedent from 'dedent';
 
 const ALLOWED_ISSUES_ACTIONS = ['opened'];
 
-const URL_REGEX = /(https?:\/\/)?([\w\-]+\.)+[\w\-]{2,}/gi;
-const EXCLUSION_LIST = [
+export const URL_REGEX = /(https?:\/\/)?([\w\-]+\.)+[\w\-]{2,}/gi;
+export const EXCLUSION_LIST = [
```
Comment (arkon) [inline src/feature/dupe-check.ts:8]: These don't seem to be used in the new feature file?

File: src/feature/existing-check.ts [hunk 3]
```diff
@@ -0,0 +1,129 @@
+import * as core from '@actions/core';
+import * as github from '@actions/github';
+import { Issue, IssuesOpenedEvent } from '@octokit/webhooks-definitions/schema';
+import axios from 'axios';
+
+import { URL_REGEX, EXCLUSION_LIST, urlsFromIssueBody } from './dupe-check';
```
Comment (arkon) [inline src/feature/existing-check.ts:6]: Could pull that out into another file, seems a bit awkward that it's shared but still in a specific feature file.

**Review by arkon — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (alessandrojean) [reply package.json:18]: Ah, oops. Didn't know the types' version was tightly coupled with the Node one.

**Pull request merged**